File: src/example/service_server.py
import rospy
from tutor.srv import TanyaJawab, TanyaJawabResponse

def handle_tanya_jawab(req):

    rospy.loginfo(f"Request: {req.tanya}")
    if req.tanya.lower() == "siapa namamu":
        return TanyaJawabResponse("Namaku ROS")
    elif req.tanya.lower() == "hi":
        return TanyaJawabResponse("Hallo")
    elif req.tanya.lower() == "hallo":
        return TanyaJawabResponse("Hi")
    else:
        return TanyaJawabResponse("Maaf, aku tidak mengerti")



if __name__ == "__main__":
    rospy.init_node("service_server")
    service_server = rospy.Service("tanya_jawab", TanyaJawab, handle_tanya_jawab)
    rospy.loginfo("Service server is ready")
    rospy.spin()

File: src/example/service_server.cpp
#include "ros/ros.h"
#include "tutor/TanyaJawab.h"
#include <boost/algorithm/string.hpp>

bool jawab(tutor::TanyaJawab::Request  &req,
         tutor::TanyaJawab::Response &res)
{
    std::string s = boost::algorithm::to_lower_copy(req.tanya);
    if (s.compare("hallo") == 0)
    {
        res.jawab = "Hi";
    }
    else if (s.compare("hi") == 0)
    {
        res.jawab = "Hallo";
    }
    else
    {
        res.jawab = "Maaf, aku tidak mengerti";
    }
  ROS_INFO("request: %s", req.tanya.c_str());
  ROS_INFO("sending back response: [%s]", res.jawab.c_str());
  return true;
}

int main(int argc, char **argv)
{
  ros::init(argc, argv, "tanya_jawab_server");
  ros::NodeHandle n;

  ros::ServiceServer service = n.advertiseService("tanya_jawab", jawab);
  ROS_INFO("Siap tanya jawab!.");
  ros::spin();
  return 0;
}